app/uk/gov/hmrc/tai/connectors/TaxAccountConnector.scala

```diff
@@ -117,13 +118,10 @@ class TaxAccountConnector @Inject()(httpHandler: HttpHandler, servicesConfig: Se
     }
 
   def updateEstimatedIncome(nino: Nino, year: TaxYear, newAmount: Int, id: Int)(
-    implicit hc: HeaderCarrier): Future[TaiResponse] =
-    httpHandler.putToApi(updateTaxCodeIncome(nino.nino, year, id), UpdateTaxCodeIncomeRequest(newAmount)) map (_ =>
-      TaiSuccessResponse) recover {
-      case e: Exception =>
-        logger.warn(s"Error while updating estimated income for $nino with exception:${e.getMessage}")
-        TaiTaxAccountFailureResponse(e.getMessage)
-    }
+    implicit hc: HeaderCarrier): Future[Done] =
+    httpHandler
+      .putToApi(updateTaxCodeIncome(nino.nino, year, id), UpdateTaxCodeIncomeRequest(newAmount))
+      .map(_ => Done)
```

Comment:
do we need to add a recover on here?

Reply:
we can but looking at where it was used... the recover was to wrap exceptions and then throw runtime ones to be caught again using NonFatal recover :D

Reply:
cool as long as it is caught higher up I am happy